### PR TITLE
[NOTEPAD] Make "UTF-8 without BOM" default

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Improve standard and usability.
JIRA issue: [CORE-18837](https://jira.reactos.org/browse/CORE-18837)

## Proposed changes

- Define a constant `ENCODING_DEFAULT` that is equal to `ENCODING_UTF8`.
- Initialize `Globals.encFile` as `ENCODING_DEFAULT`.
- Reset encoding and line ending in `DIALOG_FileNew`.
- `AnalyzeEncoding` returns `ENCODING_DEFAULT` for non-zero ASCII text.
- Remove unnecessary `DIALOG_StatusBarAlignParts` calls.

## TODO

- [x] Do tests.

### CODE REVIEW

**Review by learn-more — APPROVED** [review]

**Comment by learn-more** [comment]
Does this mean that files will be saved with a BOM by default?
If that's the case, this PR should not be merged.

**Comment by katahiromz** [comment]
> files will be saved with a BOM by default?

Nope. The default encoding is UTF-8 without BOM.

File: base/applications/notepad/main.h [hunk 1]
```diff
@@ -41,12 +41,11 @@ typedef enum
     ENCODING_UTF16LE =  1,
     ENCODING_UTF16BE =  2,
     ENCODING_UTF8    =  3,
-    ENCODING_UTF8BOM =  4
+    ENCODING_UTF8BOM =  4,
+    /* ENCODING_DEFAULT = ENCODING_ANSI, */
+    ENCODING_DEFAULT = ENCODING_UTF8,
 } ENCODING;
```
Comment (HBelusca) [inline base/applications/notepad/main.h:48]: Hm... Move "ENCODING_DEFAULT" as a #define instead ?

```suggestion
    ENCODING_UTF8BOM =  4
} ENCODING;

#define ENCODING_DEFAULT    ENCODING_UTF8 // ENCODING_ANSI

```

File: base/applications/notepad/dialog.c [hunk 2]
```diff
@@ -945,6 +944,7 @@ VOID DoShowHideStatusBar(VOID)
     ShowWindow(Globals.hStatusBar, (Globals.bShowStatusBar ? SW_SHOWNOACTIVATE : SW_HIDE));
 
     /* Update status bar contents */
+    DIALOG_StatusBarAlignParts();
```
Comment (TAN-Gaming) [inline base/applications/notepad/dialog.c:947]: This is not needed. `WM_SIZE` message above have done it already.

https://github.com/reactos/reactos/blob/9672dc50479f42c6da51b426618f05a9394a9ef2/base/applications/notepad/dialog.c#L933

https://github.com/reactos/reactos/blob/9672dc50479f42c6da51b426618f05a9394a9ef2/base/applications/notepad/main.c#L389

File: base/applications/notepad/dialog.c [hunk 3]
```diff
@@ -945,6 +944,7 @@ VOID DoShowHideStatusBar(VOID)
...
     DIALOG_StatusBarUpdateAll();
```
Comment (TAN-Gaming) [inline base/applications/notepad/dialog.c:948]: Suggestion: Add `Globals.bShowStatusBar` check as these updates are needed only if the status bar is visible.

File: base/applications/notepad/dialog.c (hunk 2 above)
Comment (katahiromz) [reply base/applications/notepad/dialog.c:947]: In order to set each part's text, the parts must be already split.

Comment (katahiromz) [reply base/applications/notepad/dialog.c:947]: I will change "posting" to "sending".

Comment (katahiromz) [reply base/applications/notepad/dialog.c]: Fixed.

**Comment by JoachimHenze** [comment]
Will this PR also help for https://jira.reactos.org/browse/CORE-18846 Katayama?

**Pull request merged**